### DISCUSSION TITLE
Re-run failed jobs, doesn't use the updated PR body

### DISCUSSION THREAD

**Discussion by tomgrossman** [community/community#48251] [Actions] [answered]
### Select Topic Area

General

### Body

I have a workflow verifying and enforcing that a PR body has a text in it, using the `${{ github.event.pull_request.body }}` expression.
When someone forget to add this "text" to the body, he edit the PR and re-run the failed job or even the entire pipeline.
However, the job doesn't use the updated PR body so it fails again.

The only way to overcome this, is by triggering the workflow again with a commit. It can be done with an empty commit like `git commit --allow-empty -m "trigger GitHub actions"` but that still sounds strange to me.

Is that a bug? Can I overcome this in a different way?

**Comment by airtower-luna** [comment, marked as answer]
You might want to add `edited` to the `types` of the `pull_request` event, so the event is triggered when the PR is edited. See https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request, the possible types and defaults are listed there.
